cmd/openfga/main_test.go

```diff
@@ -569,7 +569,130 @@ func connect(t *testing.T, tester OpenFGATester) *grpc.ClientConn {
+	resp, err := client.CreateStore(context.Background(), &openfgav1.CreateStoreRequest{
+		Name: "openfga-demo",
+	})
+	require.NoError(t, err)
+	storeID := resp.GetId()
+
+	model := parser.MustTransformDSLToProto(`model
+	schema 1.1
+type user
+
+type document
+  relations
+	define viewer: [user]
+`)
+
+	writeModelResp, err := client.WriteAuthorizationModel(context.Background(), &openfgav1.WriteAuthorizationModelRequest{
+		StoreId:         storeID,
+		TypeDefinitions: model.GetTypeDefinitions(),
+		Conditions:      model.GetConditions(),
+		SchemaVersion:   typesystem.SchemaVersion1_1,
+	})
+	require.NoError(t, err)
+	require.NotPanics(t, func() { ulid.MustParse(writeModelResp.GetAuthorizationModelId()) })
+	modelID := writeModelResp.GetAuthorizationModelId()
+
+	tests := []struct {
+		name   string
+		input  *openfgav1.WriteRequest
+		output output
+	}{
+		{
+			name: "happy-path",
+			input: &openfgav1.WriteRequest{
+				StoreId:              storeID,
+				AuthorizationModelId: modelID,
+				Writes: &openfgav1.WriteRequestWrites{
+					TupleKeys: []*openfgav1.TupleKey{
+						{Object: "document:1", Relation: "viewer", User: "user:jon"},
+					},
+				},
+			},
+			output: output{
+				errorCode: codes.OK,
+			},
+		},
+		{
+			name: "invalid-store-id",
+			input: &openfgav1.WriteRequest{
+				StoreId:              "invalid-store-id",
+				AuthorizationModelId: modelID,
+				Writes:               &openfgav1.WriteRequestWrites{},
+			},
+			output: output{
+				errorCode:    codes.InvalidArgument,
+				errorMessage: `value does not match regex pattern "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$`,
+			},
+		},
+		{
+			name: "invalid-model-id",
+			input: &openfgav1.WriteRequest{
+				StoreId:              storeID,
+				AuthorizationModelId: "invalid-model-id",
+				Writes:               &openfgav1.WriteRequestWrites{},
+			},
+			output: output{
+				errorCode:    codes.InvalidArgument,
+				errorMessage: `value does not match regex pattern "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$`,
+			},
+		},
+		{
+			name: "empty-writes",
+			input: &openfgav1.WriteRequest{
+				StoreId:              storeID,
+				AuthorizationModelId: modelID,
+				Writes:               &openfgav1.WriteRequestWrites{},
+			},
+			output: output{
+				errorCode:    codes.InvalidArgument,
+				errorMessage: `invalid WriteRequestWrites.TupleKeys: value must contain at least 1 item(s)`,
+			},
+		},
+		{
+			name: "invalid-tuple-wildcard",
+			input: &openfgav1.WriteRequest{
+				StoreId:              storeID,
+				AuthorizationModelId: modelID,
+				Writes: &openfgav1.WriteRequestWrites{
+					TupleKeys: []*openfgav1.TupleKey{
+						{Object: "document:1", Relation: "viewer", User: "user:jon"},
+						{Object: "document:1", Relation: "viewer", User: "user:*"},
+					},
+				},
+			},
+			output: output{
+				errorCode:    codes.Code(2000),
+				errorMessage: "the typed wildcard 'user:*' is not an allowed type restriction for 'document#viewer'",
```

Comment:
This logic is already tested here: https://github.com/openfga/openfga/blob/65283c2e627814c6130cd8127ed0f629895afded/internal/validation/validation_test.go#L510

I think having another one here doesn't add any value

Reply:
Certainly some redundancy between unit tests and functional test but perhaps the value is ensuring that the validation function is integrated?

Reply:
**Update:** Was explained that there was ample testing elsewhere to ensure that the validation is hooked up. Am removing this case.

Reply:
> Am removing this case.

Did you forget to commit? I still see it 🤔

Reply:
Yes, my bad 🙈. Here is the removal: https://github.com/openfga/openfga/pull/1242/commits/afc7479a21f5db89fa6ef89f62c6d52adbced818